### PR TITLE
Add thanos helm chart to stable

### PR DESCRIPTION
**Pull request by ivan-kiselev**
Signed-off-by: Ivan Kiselev <kiselev_ivan@pm.me>

Thanos project: https://github.com/improbable-eng/thanos

Hello mighty maintainers!

I tried to satisfy all the requirements for stable repo.

Not all components are going up after deploying with defaults, but there're NOTES.txt about it. 

Tell me if I missed something.

### CODE REVIEW

**Comment by homelessnessbo** [comment]
what do I do to satisfy maintainer linter?

**Comment by monotek** [comment]
Is this an "official" thanos chart by the thanos maintainers?

By the way there are already 2 Thanos charts available: 

https://hub.helm.sh/charts?q=thanos

**Comment by homelessnessbo** [comment]
@monotek 

> Is this an "official" thanos chart by the thanos maintainers?

No, it's not. Though we can ask them what are they thinking about that :) @bwplotka

> By the way there are already 2 Thanos charts available:

Thanks for pointing this out, I did not see them. I just reviewed them and I find them both much less complete and acceptable for my personal purposes. 

Though what I committed is lacking support of some things like AWS storage backend (GCS supported only) - but in general I'm actively working on it's already much more complete.

I'm sharing it and trying to make "official" because I have a feeling that in the end I'll produce something quite generic and stable (not sure how much problem or bugs this chart currently has, quite robust for me) but I'm lacking of opportunity to test things like AWS because I don't have free access to it so I wanted to involve the community somehow to produce complete generic with one-line command thanos installation.

**Comment by mazzy89** [comment]
I'm interested in this. What should we do in order to progress here?

**Comment by homelessnessbo** [comment]
@mazzy89 the best you can do is test, give feedback and contribute if something is missing at the moment I guess :)

**Comment by mazzy89** [comment]
For sure. I'll test it in my spare time over the weekend. I'll be back with some results next week.

File: stable/thanos/templates/rule-store-api-svc.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{{- with .Values.ruleLayer -}}
+{{- if .enabled -}}
+{{- if .storeApiService.enabled -}}
```
Comment (mikesplain) [inline stable/thanos/templates/rule-store-api-svc.yaml:3]: Looks like this doesn't exist in the values.yaml provided, when scoped to `.Values.ruleLayer`?  I had to copy in storeApiService from elsewhere.

**Review by mikesplain — COMMENTED** [review]
This looks like a great start to this chart. I'm going to get this running in our infra and let you know.

File: stable/thanos/templates/rule-store-api-svc.yaml (hunk 1 above)
Comment (ivan-kiselev) [reply stable/thanos/templates/rule-store-api-svc.yaml:3]: @mikesplain I fixed that, also backported changes from my local repo

**Comment by homelessnessbo** [comment]
@mikesplain do I really need to assign it to @ davidkarlsen as the bot says ?

**Comment by mikesplainsonos** [comment]
@homelessnessbo Yes, I think you need an org member to first of all mark this as `/ok-to-test` and start the testing process as we do in the kubernetes org.  You may also want to take a look at the above circleci lint.

**Comment by mikesplain** [comment]
@homelessnessbo Yes, I think you need an org member to first of all mark this as /ok-to-test and start the testing process as we do in the kubernetes org. You may also want to take a look at the above circleci lint.

**Comment by mariogeorgievft** [comment]
Is there any ETA for this PR? Would be very nice to have this helm chart at hand.

**Comment by homelessnessbo** [comment]
This PR is open for almost 3 months. I don't really know how to bring this to attention.

**Review by monotek — CHANGES_REQUESTED** [review]
first of all the lint error should be fixed.
see: https://circleci.com/gh/helm/charts/80703?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Comment by homelessnessbo** [comment]
@monotek sorry, my bad. I really did not understand what does it test maintainers list against until now :P

Seems to be fixed

**Comment by homelessnessbo** [comment]
@monotek it seems the check failed on deletion step. I don't think this is related to the chart itself, more to the testing environment, right?

**Comment by homelessnessbo** [comment]
@monotek the check is never going to succeed, one of the components require some object storage provided to start, otherwise it's going to crash. It's mentioned in `NOTES.txt`

File: stable/thanos/README.md [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+
+## TODO
+
+* Add all the storages (only GCS is supported atm)
+## Introduction
+
+This chart bootstraps the next [Thanos](https://github.com/improbable-eng/thanos) components' deployments on a [Kubernetes](http://kubernetes.io) cluster using the [Helm](https://helm.sh) package manager:
+
+* [Query](https://github.com/improbable-eng/thanos/blob/master/docs/components/query.md)
+* [Rulre](https://github.com/improbable-eng/thanos/blob/master/docs/components/Rulre.md)
```
Comment (rince86) [inline stable/thanos/README.md]: Typo.

**Comment by maorfr** [comment]
please take a look at https://github.com/helm/helm/pull/6095#issue-301608768

we strongly encourage people to self host chart repos:
https://github.com/helm/hub/blob/master/Repositories.md

you can even do that using GitHub pages:
https://github.com/helm/hub/blob/master/docs/github.md#github-pages-as-a-repository

**Comment by bwplotka** [comment]
Looks @homelessnessbo that it would be better for us to host one instead on our side @homelessnessbo  (:

**Comment by bwplotka** [comment]
Plus looks like this is already quite up-to-date: https://hub.helm.sh/charts/kiwigrid/prometheus-thanos

**Pull request closed without merging**